src/software/ai/hl/stp/play_info.h

```diff
@@ -3,22 +3,68 @@
 #include <string>
 #include <vector>
 
-struct PlayInfo
+/**
+ * A description of the play.  Contains the type of play being
+ * run, its name, and the tactic assignments for all the robots
+ * on the field.
+ */
+class PlayInfo
 {
+   public:
+    /**
+     * Constructs a PlayInfo object.
+     */
+    explicit PlayInfo();
+
+    /**
+     * Get play type
+     *
+     * @return play_type field
+     */
+    std::string getPlayType() const;
+
+    /**
+     * Get play name
+     *
+     * @return play_name field
+     */
+    std::string getPlayName() const;
+
+    /**
+     * Get vector containing tactic assignments for robots
+     *
+     * @return robot_tactic_assignment field
+     */
+    std::vector<std::string> getRobotTacticAssignment() const;
```

Comment:
Maybe we should add something here about ordering?

Thoughts anyone?

Reply:
:man_shrugging: - I'm fine with it as-is, unless there's a specific reason to order things a certain way?

Reply:
Nope, I ordering isn't important.  It was just a way for me too keep track of the variables.

Reply:
Nope. Then say it is not ordered, or should we assume that?

Reply:
Good point. I think a good way to make this clear would be to change from std::vector to [std::set](http://www.cplusplus.com/reference/set/set/).

Reply:
Hmmm I think the values were sorted before being stored in the PlayInfo struct. This was because we wanted to display things in order of robot ID in the visualizer, but because the data is simply strings it's hard to sort on the Visualizer side and requires assumptions.

I'm fine changing it to a set but can we make a ticket to change the "tactic info" in the `PlayInfo` to maybe a map of `int -> string` or a list of pairs?

Reply:
@MathewMacDougall  I think 'robot_tactic_assignment' is the "tactic info", right?  So, should I change it to a HashMap or a Set?

Reply:
That's correct. [std::unordered_set](https://en.cppreference.com/w/cpp/container/unordered_set) should be good (should have the least extra stuff to implement)

Reply:
I was thinking about the usage of the unordered_set, but I'm not too sure if it is good enough for our purposes.  Once you insert a key into the set, since it is immutable, you can't change its value from then onwards. So, if we want to change the tactic assignment for a robot, we have to delete that key first, and then add a new one.  I wonder if a HashMap suits this case better?

Reply:
Given that we only overwrite the entire list at once, I think `std::unordered_set` is fine here. Assuming that by `HashMap` you mean `std::unordered_map`, that will be the appropriate choice for when this function returns a map of `int -> string`. 

Hopefully I understood your comment correctly and that clears things up, please let me know if not though!!

Reply:
So, is there no case where we only want to change the tactic assignment of only one robot?

Reply:
Generally no. Right now once the PlayInfo is created it is generally not changed. If the Play changes, a new PlayInfo is created and sent out.

Since we don't use setting functionality for now, an `unordered_set` should be fine (TIL the keys are immutable). We can also remove the setter functions for now then since they won't serve much purpose (and YAGNI)

Reply:
For context YAGNI = "You Aren't Gonna Need It". We try avoid adding things if we don't have a use case for them yet, to reduce overall code size and reduce complexity.

https://www.martinfowler.com/bliki/Yagni.html

Reply:
Okay! Makes sense. Will make the changes soon